### ISSUE TITLE
Verify support package's Python version matches running Python version

### ISSUE THREAD

**Issue by rmartin16** [beeware/briefcase#1581]
### Describe the bug

If a user specifies a support package that contains a different version of Python than is being used to create the app, the resulting package is likely to be unusable. Errors are also likely to manifest in strange ways since the app's Python will be trying to load packages for a different version.

### Steps to reproduce

1. Activate an environment for a version other than 3.11
2. `briefcase package linux appimage --config 'requires=["toga-gtk==0.3.1","PyGobject<3.46.0"]' --config "support_package='https://github.com/indygreg/python-build-standalone/releases/download/20230507/cpython-3.10.11+20230507-x86_64_v2-unknown-linux-gnu-install_only.tar.gz'"`

If you try this with Flatpak, it errors during the build process trying to modify files for a non-existent Python....although, that still quite obscures the underlying problem for the user.

### Expected behavior

Inform the user about the incompatible settings and abort.

### Environment

- Operating System: pop os 22.04
- Python version: 3.11.6
- Software versions:
  - Briefcase: `0.3.17.dev163+g60d0a976.d20231219`

**Comment by freakboy3742** [comment]
Agreed this is a problem; detecting it could be an issue, as there's no reliable way to detect the version of a support package (that I can think of). The best I can think of would be a backend-specific identity check, looking for key filenames (like a python3.X include folder) in the unpacked support package.

The BeeWare generated support packages (for iOS/macOS) will be in a slightly better position, as there is an actual metadata file that contains this detail; Linux and Windows packages could be interesting.
